### PR TITLE
Qt 5.15.2 patches

### PR DESCRIPTION
**Pull request by rnicoll**
So I tried to run this, and then realised a few things:

* The file extension is now `.xz`, so I fixed that.
* The patches we used to use no longer are correct.

So I've copied the patches Bitcoin uses for its most recent version of Qt, onto this, to try catching up, and it's a bit... it's now a huge change set. I'm thinking we should do an intermediary Qt release.

For testing, if you change to the `depends/` folder then run `make`, that's when it actually grabs these versions.

### CODE REVIEW

File: depends/packages/qt.mk [hunk 1]
```diff
@@ -118,65 +118,91 @@ define $(package)_extract_cmds
   echo "$($(package)_qttools_sha256_hash)  $($(package)_source_dir)/$($(package)_qttools_file_name)" >> $($(package)_extract_dir)/.$($(package)_file_name).hash && \
   $(build_SHA256SUM) -c $($(package)_extract_dir)/.$($(package)_file_name).hash && \
   mkdir qtbase && \
-  tar --strip-components=1 -xf $($(package)_source) -C qtbase && \
+  tar --no-same-owner --strip-components=1 -xf $($(package)_source) -C qtbase && \
```
Comment (fmhc) [inline depends/packages/qt.mk:121]: why is that --no-same-owner flag necessary?

**Review by fmhc — COMMENTED** [review]
huh, crazy, that is quite a complex update. but i see, the upstream development with the patch system goes in here.
I had a look at https://github.com/bitcoin/bitcoin/blob/7fcf53f7b4524572d1d0c9a5fdc388e87eb02416/depends/patches/qt/no-xlib.patch and https://github.com/bitcoin/bitcoin/blob/7fcf53f7b4524572d1d0c9a5fdc388e87eb02416/depends/patches/qt/no_sdk_version_check.patch ... i will have another look at this later and try it out, will probably set up a gitian build environment first.

**Comment by rnicoll** [comment]
Going to close this as it was just to ensure you'd seen the branch, rather than it's something you should merge in.

I'm definitely happy we try moving straight to 5.15.2 BTW, just thinking you might find it easier to do it in 2-3 PRs rather than one massive jump. Very challenging untangling which patches we need and which are fine to ignore, from Bitcoin.

File: depends/packages/qt.mk (hunk 1 above)
Comment (rnicoll) [reply depends/packages/qt.mk:121]: Honestly I just copied the whole thing over from Bitcoin then tried to clean it up, then realised it wasn't going to be a small task where I could just give you the next steps.

It comes from https://github.com/bitcoin/bitcoin/commit/89bee1bdbf7f77639e053dd004629e0db0e10724 but I'm happy you ignore this (or any of the other changes in the PR), as long as the result works :)

**Pull request closed without merging**